art/attacks/evasion/feature_adversaries/feature_adversaries_pytorch.py

```diff
@@ -0,0 +1,240 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2021
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the Feature Adversaries attack in PyTorch.
+
+| Paper link: https://arxiv.org/abs/1511.05122
+"""
+import logging
+from typing import TYPE_CHECKING, Optional, Union
+
+import numpy as np
+from tqdm.auto import trange
+
+from art.attacks.attack import EvasionAttack
+from art.estimators.estimator import BaseEstimator, NeuralNetworkMixin
+
+if TYPE_CHECKING:
+    # pylint: disable=C0412
+    import torch
+    from torch.optim import Optimizer
+
+    from art.utils import CLASSIFIER_NEURALNETWORK_TYPE
+
+logger = logging.getLogger(__name__)
+
+
+class FeatureAdversariesPyTorch(EvasionAttack):
+    """
+    This class represent a Feature Adversaries evasion attack in PyTorch.
+
+    | Paper link: https://arxiv.org/abs/1511.05122
+    """
+
+    attack_params = EvasionAttack.attack_params + [
+        "optimizer",
+        "optimizer_kwargs",
+        "delta",
+        "lambda_",
+        "layer",
+        "max_iter",
+        "batch_size",
+        "step_size",
+        "random_start",
+        "verbose",
+    ]
+
+    _estimator_requirements = (BaseEstimator, NeuralNetworkMixin)
+
+    def __init__(
+        self,
+        classifier: "CLASSIFIER_NEURALNETWORK_TYPE",
```

Comment:
```suggestion
        estimator: "PyTorchEstimator",
```

Reply:
As just discussed, I added the additional `PYTORCH_ESTIMATOR_TYPE` to `art/utils.py` and using this one now :+1: